WalletWasabi/Logging/BenchmarkLogger.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace WalletWasabi.Logging
+{
+	public class BenchmarkLogger : IDisposable
+	{
+		private LogLevel LogLevel { get; }
+		public DateTimeOffset InitStart { get; }
+
+		public string OperationName { get; }
+		public string CallerFilePath { get; }
+		public int CallerLineNumber { get; }
+
+		private BenchmarkLogger(LogLevel logLevel = LogLevel.Info, [CallerMemberName]string operationName = "", [CallerFilePath] string callerFilePath = "", [CallerLineNumber] int callerLineNumber = -1)
+		{
+			LogLevel = logLevel;
+			InitStart = DateTimeOffset.UtcNow;
```

Comment:
I would use StopWatch for measuring.

Reply:
Why?

Reply:
If we go to the milliseconds' domain it is more precise.

Reply:
Done.